### DISCUSSION TITLE
Maximum size pdf support in pypdf

### DISCUSSION THREAD

**Discussion by macarneiro1** [py-pdf/pypdf#2437] [General]
Hello, I have an application server that handles pdfs and other types of documents, while developing the server that receives a base64 string a friend came up with a question about the maximum size supported and I couldn't find anything about that in the documentation. Is there any size limit to pdfs that pypdf can handle? Just because I might need to handle that just in case. Thanks in advance

**Comment by pubpub-zz** [comment]
there is no precise answer to your question
memory usage depends on what process you plan to do. What I can say is that I never came to an out of memory case. The biggest point you should worry about for me which should lead to some DoS cases

**Reply by MartinThoma** [reply]
The biggest file I've handled with pypdf was a set of PDF that I merged to a single 301MB PDF on my 16GB Laptop.

It's an interesting brain teaser to think about how to generate a PDF that would cause the maximum [peak memory usage](https://stackoverflow.com/q/33430920/562769) for our workflows:

1. Text Extraction
2. Merging
3. Metadata extraction
4. Reading Annotations
5. Image extraction
6. Compression
7. ...
